themes/default/content/blog/2022-03-22-my-pulumi-managing-my-dns/index.md

```diff
@@ -0,0 +1,249 @@
+---
+title: "My Pulumi: Managing My DNS"
+date: 2022-03-22T10:00:00Z
+draft: false
+meta_desc: In this aritcle, Rawkode shows how he using Pulumi to manage his DNS records for his many domains
+meta_image: meta.png
+authors: ["david-flanagan"]
+tags: ["Domains", "DNS"]
+---
+
+Hello, my name is David Flanagan and I own more domains than I need. The problem is, I have too many ideas; and as we all know, ideas don't become real until you buy the domain name. Unfortunately, more often than not, that's about as far as my ideas go - because, **life**. That being said, I do try to keep my DNS records under control in the event that life affords me the time to follow-up on one of these ideas. Today, I want to show you how I do that.
+
+## Overview
+
+Using Pulumi's ComponentResource API, we're able to provide a nice developer experience for manipulating DNS records through methods on an object - we call this object the "Domain Controller". I have a domain controller for every domain that I own. Through this controller I can add DNS records through helper methods that keep the interface as simple as can be. Let's take a look.
```

Comment:
```suggestion
Using Pulumi's ComponentResource API, we're able to provide a nice developer experience for manipulating DNS records through methods on an object---we call this object the "Domain Controller." I have a domain controller for every domain that I own. Through this controller I can add DNS records through helper methods that keep the interface as simple as can be. Let's explore it.
```

Reply:
I think I'd put a link here (maybe https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/#ComponentResource ?) as someone may wonder what package you're using since it's hard to search for.

Reply:
+1!